From 6fac2cdbc34eb2fa4a36d38d51779b23e71032b8 Mon Sep 17 00:00:00 2001
From: Chen Xu <fionaxu@google.com>
Date: Thu, 24 Jun 2021 11:08:00 -0700
Subject: [PATCH 08/11] [Q_asb_2021-10] fix the issue that clearCallingIdentity
 before appops check

we should restore CallingIndentity when do the appops check for
access device/subscriber identifier

Bug: 187147737
Bug: 183612370
Test: Manual test
Change-Id: Id0abfee602823f56811799a6d5c2bbe8cd5e2cc1
(cherry picked from commit 020d831ced2393aca3b51eb2a9fb805f5757fd87)
Merged-in: Id0abfee602823f56811799a6d5c2bbe8cd5e2cc1
(cherry picked from commit bb73fd02340e8764e236006012f77fd5245d549c)
---
 .../telephony/SubscriptionController.java     | 22 +++++++++----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/java/com/android/internal/telephony/SubscriptionController.java b/src/java/com/android/internal/telephony/SubscriptionController.java
index da33f1d73e..8de8a73fe1 100644
--- a/src/java/com/android/internal/telephony/SubscriptionController.java
+++ b/src/java/com/android/internal/telephony/SubscriptionController.java
@@ -730,22 +730,22 @@ public class SubscriptionController extends ISub.Stub {
 
         // Now that all security checks passes, perform the operation as ourselves.
         final long identity = Binder.clearCallingIdentity();
+        List<SubscriptionInfo> subList;
         try {
-            List<SubscriptionInfo> subList = null;
             subList = getSubInfo(null, null);
-            if (subList != null) {
-                if (VDBG) logd("[getAllSubInfoList]- " + subList.size() + " infos return");
-                subList.stream().map(
-                        subscriptionInfo -> conditionallyRemoveIdentifiers(subscriptionInfo,
-                                callingPackage, "getAllSubInfoList"))
-                        .collect(Collectors.toList());
-            } else {
-                if (VDBG) logd("[getAllSubInfoList]- no info return");
-            }
-            return subList;
         } finally {
             Binder.restoreCallingIdentity(identity);
         }
+        if (subList != null) {
+            if (VDBG) logd("[getAllSubInfoList]- " + subList.size() + " infos return");
+            subList.stream().map(
+                    subscriptionInfo -> conditionallyRemoveIdentifiers(subscriptionInfo,
+                            callingPackage, "getAllSubInfoList"))
+                    .collect(Collectors.toList());
+        } else {
+            if (VDBG) logd("[getAllSubInfoList]- no info return");
+        }
+        return subList;
     }
 
     private List<SubscriptionInfo> makeCacheListCopyWithLock(List<SubscriptionInfo> cacheSubList) {
-- 
2.35.1

